crypto/async/arch/async_posix.h

```diff
@@ -25,17 +25,27 @@
 #  define ASYNC_POSIX
 #  define ASYNC_ARCH
 
+#  ifdef __CET__
+#    define USE_SWAPCONTEXT
+#  endif
 #  include <ucontext.h>
-#  include <setjmp.h>
+#    ifndef USE_SWAPCONTEXT
+#      include <setjmp.h>
+#    endif
 
 typedef struct async_fibre_st {
     ucontext_t fibre;
+#    ifndef USE_SWAPCONTEXT
     jmp_buf env;
     int env_init;
+#    endif
 } async_fibre;
 
 static ossl_inline int async_fibre_swapcontext(async_fibre *o, async_fibre *n, int r)
 {
+#    ifdef USE_SWAPCONTEXT
+    swapcontext(&o->fibre, &n->fibre);
```

Comment:
can you explain why this is done?
maybe even add a comment ?

Reply:
I will add some comments.